src/client.ts

```diff
@@ -387,6 +387,8 @@ export interface ICreateClientOpts {
      * Default: false.
      */
     isVoipWithNoMediaAllowed?: boolean;
+
+    logger?: PrefixedLogger;
```

Comment:
I suppose this needs documentation, I'm not sure if it also needs tests?

Reply:
It might need tests just to pass the quality gate, I'm not really sure what to test here though? Have two matrix clients, log to them and ensure they go to the correct logger? Seems almost like testing the JavaScript engine itself though...

Reply:
That makes sense, we should ensure this doesn't break. 👍